apps/prairielearn/src/pages/publicAssessments/publicAssessments.html.ts

```diff
@@ -2,18 +2,90 @@ import { html } from '@prairielearn/html';
 
 import { AssessmentModuleHeading } from '../../components/AssessmentModuleHeading.html.js';
 import { AssessmentSetHeading } from '../../components/AssessmentSetHeading.html.js';
+import { Modal } from '../../components/Modal.html.js';
 import { PageLayout } from '../../components/PageLayout.html.js';
+import type { CopyTarget } from '../../lib/copy-content.js';
 import type { CourseInstance } from '../../lib/db-types.js';
 import { type AssessmentRow } from '../../models/assessment.js';
 
+function CopyCourseInstanceModal({
+  courseInstance,
+  courseInstanceCopyTargets,
+}: {
+  courseInstance: CourseInstance;
+  courseInstanceCopyTargets: CopyTarget[] | null;
+}) {
+  if (courseInstanceCopyTargets == null) return '';
+  return Modal({
+    id: 'copyCourseInstanceModal',
+    title: 'Copy course instance',
+    formAction: courseInstanceCopyTargets[0]?.copy_url ?? '',
+    formClass: 'js-copy-question-form',
+    form: courseInstanceCopyTargets?.length > 0,
+    body:
+      courseInstanceCopyTargets.length === 0
+        ? html`
+            <p>
+              You can't copy this course instance because you don't have editor permissions in any
+              courses.
+              <a href="/pl/request_course">Request a course</a> if you don't have one already.
+              Otherwise, contact the owner of the course you expected to have access to.
+            </p>
+          `
+        : html`
```

Comment:
This is mostly copy-pasted from the copy question modal. 

I thought about making a 'copy content' component to generate both modals. The reason I didn't, is because the question copy modal is going to stay the same, whereas the course instance one is going to change over time as we add more options for different types of copying (copy vs. reference the questions).